test/functional/http/post/4.multisig.js

```diff
@@ -0,0 +1,530 @@
+'use strict';
+
+var node = require('../../../node');
+var shared = require('../../shared');
+var constants = require('../../../../helpers/constants');
+
+var sendTransactionPromise = require('../../../common/apiHelpers').sendTransactionPromise;
+var creditAccountPromise = require('../../../common/apiHelpers').creditAccountPromise;
+var sendSignaturePromise = require('../../../common/apiHelpers').sendSignaturePromise;
+var waitForConfirmations = require('../../../common/apiHelpers').waitForConfirmations;
+
+describe('POST /api/transactions (type 4) register multisignature', function () {
+
+	var scenarios = {
+		'no_funds': new shared.multisigScenario(3, 0),
+		'minimal_funds': new shared.multisigScenario(3, constants.fees.multisignature * 3),
+		'minimum_not_reached': new shared.multisigScenario(4), //4 members 2 min signatures required
+		'regular': new shared.multisigScenario(3), //3 members 2 min signatures required
+		'max_signatures': new shared.multisigScenario(constants.multisigConstraints.keysgroup.maxItems + 1), //16 members 2 min signatures required 
+		'max_signatures_max_min': new shared.multisigScenario(constants.multisigConstraints.keysgroup.maxItems + 1), //16 members 16 min signatures required
+		'more_than_max_signatures': new shared.multisigScenario(constants.multisigConstraints.keysgroup.maxItems + 2) //17 members 2 min signatures required
+	};
+
+	var transaction, signature;
+	var transactionsToWaitFor = [];
+	var badTransactions = [];
+	var goodTransactions = [];
+	var badTransactionsEnforcement = [];
+	var goodTransactionsEnforcement = [];
+	var pendingMultisignatures = [];
+
+	before(function () {
+		//Crediting accounts
+		return node.Promise.all(Object.keys(scenarios).map(function (key) {
+			if (key === 'no_funds') {
+				return;
+			}
+			return creditAccountPromise(scenarios[key].account.address, scenarios[key].amount).then(function (res) {
+				node.expect(res).to.have.property('success').to.be.ok;
+				node.expect(res).to.have.property('transactionId').that.is.not.empty;
+				transactionsToWaitFor.push(res.transactionId);
+			});
+		}))
+			.then(function () {
```

Comment:
Would be better I think if we have `.then` on the same line as well.
Like this: 
```
})).then(function () {
  ...
});
```